### ISSUE TITLE
Bug: `transformHandler.call({} as any, ...)` with `{}` cause the incompatible exception from `vite-plugin-inspect` with Vite 6

### ISSUE THREAD

**Issue by nekomeowww** [unocss/unocss#4296]
### UnoCSS version

0.64.1

### Describe the bug

Currently, when running projects with Vite 6, UnoCSS, and `vite-plugin-inspect` with `build: true` option is on, build will fail because of:

> Environment is required

```shell
[unocss:global:build:scan] Environment is required
[unocss:global:build:scan] Environment is required
    at InspectContext.getEnvContext (file:///Users/neko/Git/nolebase/integrations/node_modules/.pnpm/vite-plugin-inspect@0.10.0_rollup@3.29.5_vite@6.0.1_@types+node@22.10.0_jiti@2.3.3_less@4.2.1_tsx@4.19.1_/node_modules/vite-plugin-inspect/dist/index.mjs:974:13)
    at file:///Users/neko/Git/nolebase/integrations/node_modules/.pnpm/vite-plugin-inspect@0.10.0_rollup@3.29.5_vite@6.0.1_@types+node@22.10.0_jiti@2.3.3_less@4.2.1_tsx@4.19.1_/node_modules/vite-plugin-inspect/dist/index.mjs:1219:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.renderChunk (file:///Users/neko/Git/nolebase/integrations/node_modules/.pnpm/@unocss+vite@0.64.1_rollup@3.29.5_vite@6.0.1_@types+node@22.10.0_jiti@2.3.3_less@4.2.1_tsx@4._dv6kejfz7zs3fexjqdo4mnnkom/node_modules/@unocss/vite/dist/index.mjs:649:11)
    at async transformChunk (file:///Users/neko/Git/nolebase/integrations/node_modules/.pnpm/rollup@4.25.0/node_modules/rollup/dist/es/shared/node-entry.js:18848:16)
    at async file:///Users/neko/Git/nolebase/integrations/node_modules/.pnpm/rollup@4.25.0/node_modules/rollup/dist/es/shared/node-entry.js:18922:17
    at async Promise.all (index 1)
    at async transformChunksAndGenerateContentHashes (file:///Users/neko/Git/nolebase/integrations/node_modules/.pnpm/rollup@4.25.0/node_modules/rollup/dist/es/shared/node-entry.js:18917:5)
    at async renderChunks (file:///Users/neko/Git/nolebase/integrations/node_modules/.pnpm/rollup@4.25.0/node_modules/rollup/dist/es/shared/node-entry.js:18826:151)
    at async Bundle.generate (file:///Users/neko/Git/nolebase/integrations/node_modules/.pnpm/rollup@4.25.0/node_modules/rollup/dist/es/shared/node-entry.js:19084:13)
```

The problematic lines that causes the issue is:

https://github.com/unocss/unocss/blob/bc36345cc09506c54c4d057fb9d6acba5cabb110/packages/vite/src/modes/global/build.ts#L179-L188

I dived into the debugger to confirm the issue, found that when `vite-plugin-inspect` is loaded, and hijack-ed `transform` hook, since this part of the code from https://github.com/antfu-collective/vite-plugin-inspect/blob/160ee041ddc489f3d3c986a5b4416c86f4060bbd/src/node/hijack.ts#L63-L99 will wrap and replace `transform` hook from UnoCSS:

```typescript
export function hijackPlugin(...) {
  // ...
  hijackHook(plugin, 'transform', async (fn, context, args, order) => {
    // ...
    const result = error ? '[Error]' : (typeof _result === 'string' ? _result : _result?.code)
    if (ctx.filter(id)) {
      const sourcemaps = typeof _result === 'string' ? null : _result?.map
      ctx
        .getEnvContext(context.environment)
        .recordTransform(id, {
          name: plugin.name,
          result,
          start,
          end,
          order,
          sourcemaps,
          error: error ? parseError(error) : undefined,
        }, code)
    }

    if (error)
      throw error

    return _result
  })
  // ...
}
```

with the `context` remains `{}`, there is no `context['environment']` present, therefore, when `ctx.getEnvContext(context.environment)` gets called, the code will run into https://github.com/antfu-collective/vite-plugin-inspect/blob/160ee041ddc489f3d3c986a5b4416c86f4060bbd/src/node/context.ts#L58-L63 which throws the error `Environment is required` out because of the `undefined` of `env` parameter:

```typescript
export class InspectContext {
  // ...
  getEnvContext(env?: Environment) {
    if (!env)
      throw new Error('Environment is required')
    const vite = this.getViteContext(env.getTopLevelConfig())
    return vite.getEnvContext(env)
  }
  // ...
}
```

I saw that the #4289 was merged, and it just bumpped the versions of `vite`, I suspect there is no clue shown that other migration changes to Vite 6 is happening or being working on currently.

While this seems adapt-able from `vite-plugin-inspect` side (perhaps), but with the new experimenting [Environment API](https://vite.dev/guide/api-environment.html) starting from Vite 6, UnoCSS should not just pass the empty object `{}` into transform hooks when there might be some environment specific context required by the other plugins, such as `vite-plugin-inspect`.

Currently, this does have workaround, where I can simply toggle off the `build: true` when there is no need to.

### Debugging views

<img width="2559" alt="image" src="https://github.com/user-attachments/assets/22b5444a-b0fd-4faa-9ed4-de4d41f8c4ac">

<details>
  <summary>Closer look of `transformHandler`</summary>
  <p>
    <img width="776" alt="image" src="https://github.com/user-attachments/assets/d26766c6-f643-4f97-9b2b-aea505195e8b">
  </p>
</details>

<details open>
  <summary>Debugger view</summary>
  <p>
    <img width="848" alt="image" src="https://github.com/user-attachments/assets/4ba5ab80-52cb-4014-91d2-50a237715f08">
  </p>
</details>

<details>
  <summary>Debugger view</summary>
  <p>
    <img width="459" alt="image" src="https://github.com/user-attachments/assets/cd47b546-0e1f-4cf6-b1dd-bab25d233d04">
  </p>
</details>

### Reproduction

https://stackblitz.com/edit/unocss-unocss-aftlug?file=package.json

### System Info

```
  System:
    OS: macOS 14.3.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 85.88 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 22.11.0 - ~/.volta/tools/image/node/22.11.0/bin/node
    Yarn: 1.22.19 - ~/.volta/tools/image/yarn/1.22.19/bin/yarn
    npm: 10.9.0 - ~/.volta/tools/image/node/22.11.0/bin/npm
    pnpm: 9.14.2 - ~/.volta/bin/pnpm
    bun: 1.0.36 - /opt/homebrew/bin/bun
  Browsers:
    Chrome: 131.0.6778.86
    Chrome Canary: 133.0.6863.0
    Safari: 17.3.1
  npmPackages:
    vite: catalog: => 6.0.1
```

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/unocss/unocss/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't [already an issue](https://github.com/unocss/unocss/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/unocss/unocss/discussions) or join our [Discord Chat Server](https://chat.antfu.me).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by nekomeowww** [comment]
I was wrong, toggle off `build: true` into `build: false` doesn't make any difference. So the only workaround is to disable `vite-plugin-inspect` temporarily.

Perhaps, I should raise another PR to make `vite-plugin-inspect` not to hijack the hooks during `build` (maybe Build Environment also)?

**Comment by antfu** [comment]
Should be fixed with `vite-plugin-inspect@0.10.1`. Thanks for the detailed report!
